From cb53488d286e165acf422f98a9cb8452902ef577 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Wed, 25 Sep 2019 23:34:37 +0800
Subject: [PATCH 126/128] drm/lima: allow to retry when allocating memory for
 BO

Currently, when allocating the memory for BO by Mesa, the lima kernel
driver set only GFP_DMA32 flag; and this allocation may fail when the
memory is relatively adequate, thus retrying is needed.

Add the GFP flags for retrying memory allocation.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/gpu/drm/lima/lima_object.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/lima/lima_object.c b/drivers/gpu/drm/lima/lima_object.c
index 5c41f859a72f..b89011ecbe59 100644
--- a/drivers/gpu/drm/lima/lima_object.c
+++ b/drivers/gpu/drm/lima/lima_object.c
@@ -94,7 +94,9 @@ struct lima_bo *lima_bo_create(struct lima_device *dev, u32 size,
 			goto err_out;
 		}
 	} else {
-		mapping_set_gfp_mask(bo->gem.filp->f_mapping, GFP_DMA32);
+		mapping_set_gfp_mask(bo->gem.filp->f_mapping,
+				     GFP_DMA32 | __GFP_RETRY_MAYFAIL |
+				     __GFP_NOWARN);
 		bo->pages = drm_gem_get_pages(&bo->gem);
 		if (IS_ERR(bo->pages)) {
 			ret = ERR_CAST(bo->pages);
-- 
2.23.0

